src/main/java/com/powsybl/openloadflow/network/impl/LfLegBranch.java

```diff
@@ -133,10 +155,20 @@ public void updateState() {
         leg.getTerminal().setP(p.eval() * PerUnit.SB);
         leg.getTerminal().setQ(q.eval() * PerUnit.SB);
 
-        if (phaseControl != null) {
+        if (isPhaseController()) {
```

Comment:
@floriand-e2r Do not forget to add condition && phaseControl.getMode() == PhaseControl.Mode.OFF